client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-pattern-pages.ts

```diff
@@ -82,18 +83,34 @@ const usePatternPages = (
 		pageSlugs = INITIAL_PAGES;
 	}
 
-	pages = pageSlugs.map( ( slug ) => pagesMapByCategory[ slug ]?.[ 0 ] ).filter( Boolean );
+	pages = pageSlugs
+		.map( ( slug ) => {
+			const firstPage = pagesMapByCategory[ slug ]?.[ 0 ];
+			if ( firstPage ) {
+				return {
+					...firstPage,
+					title: getPagePatternTitle( firstPage ),
+				};
+			}
+		} )
+		.filter( Boolean ) as Pattern[];
+
+	pagesToShow = pageCategoriesInOrder
+		.map( ( category: Category ) => {
+			const { name } = category;
+			const hasPages = name && pagesMapByCategory[ name ]?.length;
+			if ( hasPages ) {
+				const firstPage = pagesMapByCategory[ name ][ 0 ];
+				return {
+					name,
+					hasPages,
+					title: getPagePatternTitle( firstPage ),
```

Comment:
This is required to fix the issue list of page names on the sidebar.

<img width="248" alt="Screenshot 2567-01-08 at 16 27 13" src="https://github.com/Automattic/wp-calypso/assets/1881481/9e98cae4-c15b-4802-b7ec-251fc1eb2f6d">